src/ngraph/runtime/cpu/pass/cpu_layout.cpp

```diff
@@ -990,8 +990,21 @@ namespace ngraph
                     }
                     catch (const mkldnn::error& e)
                     {
-                        throw ngraph_error("MKLDNN Unsupported pooling layout" +
-                                           to_string(input_desc.data.format) + e.message);
+                        if (arg0_shape.size() == 4 || arg0_shape.size() == 5)
+                        {
+                            auto default_format = arg0_shape.size() == 4
+                                                      ? mkldnn::memory::format::nchw
+                                                      : mkldnn::memory::format::ncdhw;
+                            auto default_desc = mkldnn_utils::create_default_mkldnn_md(
+                                node.get(), 0, false, default_format);
+                            i_mds.push_back(default_desc);
+                            o_mds.push_back(default_desc);
```

Comment:
I would move that code out of the catch block as a special-handling in the try block.

Reply:
didn't quite follow your comment. We only need this if MKLDNN doesn't support the incoming layout and the only way to know that for now is to try and create a primitive descriptor.